src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -181,8 +185,8 @@ public partial class HubConnection
         /// <remarks>
         /// The <see cref="IServiceProvider"/> used to initialize the connection will be disposed when the connection is disposed.
         /// </remarks>
-        public HubConnection(IConnectionFactory connectionFactory, IHubProtocol protocol, IServiceProvider serviceProvider, ILoggerFactory loggerFactory, IRetryPolicy reconnectPolicy)
-            : this(connectionFactory, protocol, serviceProvider, loggerFactory)
+        public HubConnection(IConnectionFactory connectionFactory, IHubProtocol protocol, EndPoint endPoint, IServiceProvider serviceProvider, ILoggerFactory loggerFactory, IRetryPolicy reconnectPolicy)
```

Comment:
I assume keeping the original constructors (to avoid the breaking change) is the entire point of this change.

Reply:
Users _shouldn't_ be constructing `HubConnection`. If it weren't for DI this would be `internal`. I'm kinda 🤷‍♂ on breaking changes in this kind of thing.

Reply:
Funnily enough, HubConnection has always been difficult/impossible to construct manually. Now that the HubConnection needs to specify an EndPoint that previously was just an implementation detail of IConnectionFactory, HubConnection has to break at least a little.

The main thing I was worried about was keeping the HubConnectionBuilder backwards-compatible.

Reply:
Stupid question, can we make HubConnection internal?

Reply:
We can't make the class internal as it's the class users interact with to send and receive messages. We would like the constructors to be internal but like [Andrew said](https://github.com/aspnet/AspNetCore/pull/11484/files#r296904380) we can't because of DI.